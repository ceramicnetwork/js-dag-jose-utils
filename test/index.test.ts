import * as u8a from 'uint8arrays'
import CID from 'cids'
import {
  encodePayload,
  prepareCleartext,
  decodeCleartext,
  encodeIdentityCID,
  decodeIdentityCID
} from '../src/index'

describe('dag-jose-utils', () => {

  it('Properly encode payload', async () => {
    const payload1 = { abc: 123 }
    const enc1 = await encodePayload(payload1)
    expect(enc1).toMatchSnapshot()
    const payload2 = { aLink: enc1.cid }
    expect(await encodePayload(payload2)).toMatchSnapshot()
  })

  it('Encodes and decodes identity CIDs', async () => {
    const ct1 = { abc: 123 }
    const enc1 = encodeIdentityCID(ct1)
    expect(enc1).toMatchSnapshot()
    expect(decodeIdentityCID(enc1)).toEqual(ct1)
    const ct2 = { aLink: enc1 }
    const enc2 = encodeIdentityCID(ct2)
    expect(enc2).toMatchSnapshot()
    expect(decodeIdentityCID(enc2).aLink.bytes).toEqual(ct2.aLink.bytes)
  })

  it('Prepare and decode cleartext', async () => {
    const ct1 = { abc: 123 }
    const enc1 = prepareCleartext(ct1)
    expect(enc1).toMatchSnapshot()
    expect(decodeCleartext(enc1)).toEqual(ct1)
    const ct2 = { aLink: enc1 }
    const enc2 = prepareCleartext(ct2)
    expect(enc2).toMatchSnapshot()
    expect(decodeCleartext(enc2).aLink.bytes).toEqual(ct2.aLink.bytes)
  })
})
