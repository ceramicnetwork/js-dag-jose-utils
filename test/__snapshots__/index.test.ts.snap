// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dag-jose-utils Encodes and decodes identity CIDs 1`] = `
Object {
  "codec": "dag-cbor",
  "hash": Uint8Array [
    0,
    7,
    161,
    99,
    97,
    98,
    99,
    24,
    123,
  ],
  "version": 1,
}
`;

exports[`dag-jose-utils Encodes and decodes identity CIDs 2`] = `
Object {
  "codec": "dag-cbor",
  "hash": Uint8Array [
    0,
    22,
    161,
    101,
    97,
    76,
    105,
    110,
    107,
    216,
    42,
    76,
    0,
    1,
    113,
    0,
    7,
    161,
    99,
    97,
    98,
    99,
    24,
    123,
  ],
  "version": 1,
}
`;

exports[`dag-jose-utils Prepare and decode cleartext 1`] = `
Uint8Array [
  1,
  113,
  0,
  7,
  161,
  99,
  97,
  98,
  99,
  24,
  123,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;

exports[`dag-jose-utils Prepare and decode cleartext 2`] = `
Uint8Array [
  1,
  113,
  0,
  33,
  161,
  101,
  97,
  76,
  105,
  110,
  107,
  88,
  24,
  1,
  113,
  0,
  7,
  161,
  99,
  97,
  98,
  99,
  24,
  123,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;

exports[`dag-jose-utils Properly encode payload 1`] = `
Object {
  "cid": Object {
    "codec": "dag-cbor",
    "hash": Uint8Array [
      18,
      32,
      62,
      231,
      147,
      209,
      63,
      207,
      7,
      204,
      121,
      70,
      28,
      81,
      63,
      219,
      19,
      21,
      11,
      57,
      9,
      225,
      136,
      89,
      176,
      179,
      28,
      77,
      95,
      227,
      241,
      26,
      49,
      202,
    ],
    "version": 1,
  },
  "linkedBlock": Uint8Array [
    161,
    99,
    97,
    98,
    99,
    24,
    123,
  ],
}
`;

exports[`dag-jose-utils Properly encode payload 2`] = `
Object {
  "cid": Object {
    "codec": "dag-cbor",
    "hash": Uint8Array [
      18,
      32,
      95,
      2,
      189,
      189,
      162,
      177,
      207,
      186,
      234,
      10,
      197,
      153,
      107,
      7,
      16,
      34,
      110,
      198,
      34,
      29,
      95,
      36,
      156,
      70,
      152,
      200,
      156,
      238,
      157,
      204,
      50,
      32,
    ],
    "version": 1,
  },
  "linkedBlock": Uint8Array [
    161,
    101,
    97,
    76,
    105,
    110,
    107,
    216,
    42,
    88,
    37,
    0,
    1,
    113,
    18,
    32,
    62,
    231,
    147,
    209,
    63,
    207,
    7,
    204,
    121,
    70,
    28,
    81,
    63,
    219,
    19,
    21,
    11,
    57,
    9,
    225,
    136,
    89,
    176,
    179,
    28,
    77,
    95,
    227,
    241,
    26,
    49,
    202,
  ],
}
`;
